proposals/roles.md

```diff
@@ -0,0 +1,665 @@
+﻿# Roles
+
+This proposal is divided into three parts, all relating to extending existing types:  
+A. static extensions  
+B. roles and extensions with members  
+C. roles and extensions that implement interfaces  
+
+The syntax for roles is as follows:
+
+```antlr
+type_declaration
+    : role_declaration // add
+    | extension_declaration // add
+    | ...
+    ;
+
+role_declaration
+    : modifier* 'role' identifier type_parameter_list? role_underlying_type? type_parameter_constraints_clause* role_body
+    ;
+
+extension_declaration
+    : modifier* 'extension' identifier type_parameter_list? role_underlying_type? type_parameter_constraints_clause* role_body
+    ;
+
+role_underlying_type
+    : ':' type
+    : ':' type ',' role_or_interface_type_list
+    ;
+
+role_or_interface_type_list
```

Comment:
I am having some trouble with this syntax. Will interfaces and roles be mixed or will roles follow interfaces?

I am uncomfortable with them being mixed.